crates/datafusion_ext/src/planner/relation/mod.rs

```diff
@@ -235,12 +235,18 @@ impl<'a, S: AsyncContextProvider> SqlQueryPlanner<'a, S> {
 /// Returns a reference to table func by inferring which function to use from a
 /// given path.
 fn infer_func_for_file(path: &str) -> Result<OwnedTableReference> {
-    let ext = Path::new(path)
-        .extension()
-        .ok_or_else(|| DataFusionError::Plan(format!("missing file extension: {path}")))?
-        .to_str()
-        .ok_or_else(|| DataFusionError::Plan(format!("strange file extension: {path}")))?
-        .to_lowercase();
+    let ext = if path.contains("docs.google.com/spreadsheet") && path.trim().ends_with("output=csv")
```

Comment:
Should we instead pass the `path` as `&Url` and check if there is a query parameter output with value `csv`? `path` ending with `output=csv` feels wrong...

Reply:
It does feel wrong. 

I also wonder if we can change the content (mime?) type header without specifying an argument?

I guess also we can do some amount of heroics if we get a URL that _doesn't_ have the argument, we could add it?